WordPress/gutenberg
# Tip: Covert component to TypeScript

## What?
Converts the `Tip` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactors the current `Tip` to TypeScript.
- Refactors the storybook to TypeScript and removed knobs.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `Tip` continues to function as expected
3. Test within the Gutenberg editor
